### PR TITLE
[SDK-519] Updating ABKIDFADelegate documentation with iOS 14 changes.

### PR DESCRIPTION
**Pull request by tbelote**
# IDFA Delegate changes and App Tracking Transparency for iOS 14 SDK-519

**Description of Change:**
Updating documentation to reflect changes to IDFA and for the new App Tracking Transparency framework required by iOS 14.

### Is this change associated with a Braze feature/product release?
- [X] Yes Appboy iOS SDK 3.27.0-Beta2 and won't be generally available until iOS 14 GM is released.

---
---

<details>
<summary>PR Checklist</summary>

- [ ] Check that all links work!
- [ ] Ensure you have completed [our CLA](https://www.braze.com/docs/cla/).
- [ ] Tag @Timothy-Kim and @KellieHawks as a reviewer when your work is _done and ready to be reviewed for merge_.
- [ ] Tag others as Reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `config/nginx.conf.erb`.

</details>

<details>
<summary>Writing & Style Quick Tips</summary>

- Use the formatting shown on the [Styling Test Page](https://github.com/Appboy/braze-docs/blob/develop/_docs/_home/styling_test_page.md) to create unique templates and vehicles for your information.
- Use Action verbs!
- Capitalize proper nouns, unique Braze tools and features, custom Braze channels, and all Header titles.
- Redirect when necessary!
- Always speak to the Braze customer - refer to them as “you”, whether they are an Engineer/Developer, a Marketer, or an Admin.
- On Voice: You should aim to be a Friendly Professor. Imagine, if you will, the smartest, but also kindest professor you had. You know, the one who showed you your potential and tested you, but also went slowly in class so you could take great notes and learn some things? Don't be too peppy, but don't be formal. You’re not selling anything - you’re instructing and informing and enabling.

_Consult the [Docs Text Formatting Guide](https://github.com/Appboy/braze-docs/wiki/Special-Formatting) and the [Docs Writing Style Guide & Best Practices](https://github.com/Appboy/braze-docs/wiki/Writing-Style-Guide-&-Best-Practices) for more details._

</details>

<details>
<summary>ATTENTION: For PR Reviewers</summary>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Preview all changes in the linked Heroku environment (click `View deployment` button below, then "Docs". A `502` error just means you should refresh until you see the Docs Home page.
</details>

-

### CODE REVIEW

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md [hunk 1]
```diff
@@ -55,17 +65,22 @@ Follow these steps to implement IDFA Collection:
...
 
-- (BOOL)isAdvertisingTrackingEnabled {
+- (BOOL)isAdvertisingTrackingEnabledOrATTAuthorized {
+  if (@available(iOS 14, *)) {
+    return [ATTrackingManager trackingAuthorizationStatus] == ATTrackingManagerAuthorizationStatusAuthorized;
+  }
   return [[ASIdentifierManager sharedManager] isAdvertisingTrackingEnabled];
 }
 
+
```
Comment (hokstuff) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:83]: nit: remove newline

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md [hunk 2]
```diff
@@ -75,18 +90,23 @@ Follow these steps to implement IDFA Collection:
...
-  func isAdvertisingTrackingEnabled() -> Bool {
+  func isAdvertisingTrackingEnabledOrATTAuthorized() -> Bool {
+    if #available(iOS 14, *) {
+      return ATTrackingManager.trackingAuthorizationStatus ==  ATTrackingManager.AuthorizationStatus.authorized
+    }
     return ASIdentifierManager.shared().isAdvertisingTrackingEnabled
   }
 
 }
+
```
Comment (hokstuff) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:109]: nit: remove newline

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md [hunk 3]
```diff
@@ -43,6 +43,16 @@ As a result, we recommend continuing to collect the IDFA if you meet any of the
 - You are attributing app installation to a previously served advertisement
 - You are attributing an action within the application to a previously served advertisement
 
+## iOS 14 AppTrackingTransparency
+Beginning with iOS 14 access to the advertising identifier is now on a per app basis. In addition to this delegate, your application will need to prompt for this authorization. For more information you can see this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency).
```
Comment (hokstuff) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:47]: Added some commas/hyphens and slight wording updates

```suggestion
Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to this delegate, your application will need to prompt for this authorization. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview on Privacy](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency).
```

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md [hunk 4]
```diff
@@ -43,6 +43,16 @@ As a result, we recommend continuing to collect the IDFA if you meet any of the
 - You are attributing app installation to a previously served advertisement
 - You are attributing an action within the application to a previously served advertisement
 
+## iOS 14 AppTrackingTransparency
+Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to this delegate, your application will need to prompt for this authorization. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview on Privacy](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency).
```
Comment (chshapiro) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:47]: `this delegate` is a little confusing and vague

Comment (chshapiro) [reply _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:47]: also will the iOS 14 guide be a thing once iOS 14 is released (@davidbielik )? we should probably just put the info here

Comment (hokstuff) [reply _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:47]: Based off Chava's comment, here's a rewording of the second sentence:

```suggestion
Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. 
Your application will need to prompt the user for authorization to track them as well as implement additional code to handle this. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview on Privacy](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency).
```

Comment (tbelote) [reply _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:47]: I'm also thinking of going through and clarify that it's really Xcode 11 vs 12

File: _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md [hunk 5]
```diff
@@ -43,6 +43,16 @@ As a result, we recommend continuing to collect the IDFA if you meet any of the
 - You are attributing app installation to a previously served advertisement
 - You are attributing an action within the application to a previously served advertisement
 
+## iOS 14 AppTrackingTransparency
+Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to implementing Braze's `ABKIDFADelegate` protocol, your application will need to request authorization using Apple's `ATTrackingManager` in the App Tracking Transparency framework. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency). With iOS 14 collecting IDFA will require building with Xcode 12, collecting IDFA wil Xcode 11 is not possible in iOS 14.
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:47]: > With iOS 14 collecting IDFA will require building with Xcode 12, collecting IDFA wil Xcode 11 is not possible in iOS 14.

This is really well summarized, going to add this to the upgrade guide, too.

Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/ios/initial_sdk_setup/other_sdk_customizations.md:47]:
```suggestion
Beginning with iOS 14, access to the advertising identifier is now on a per-app basis. In addition to implementing Braze's `ABKIDFADelegate` protocol, your application will need to request authorization using Apple's `ATTrackingManager` in the App Tracking Transparency framework. For more information, please reference this [Braze iOS 14 guide](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/ios_14/#idfa-and-app-tracking-transparency), [Apple's Overview](https://developer.apple.com/app-store/user-privacy-and-data-use/), and [Apple's Developer Documentation](https://developer.apple.com/documentation/apptrackingtransparency). With iOS 14 collecting IDFA will require building with Xcode 12, collecting IDFA will Xcode 11 is not possible in iOS 14.
```

**Comment by davidbielik** [comment]
hey @tbelote does this page also need updating or is it not affected?

1. https://www.braze.com/docs/developer_guide/platform_integration_guides/cordova/ios/initial_sdk_setup/#configure-the-plugin

**Comment by tbelote** [comment]
@KellieHawks I'm ready to merge when you're ready

**Pull request merged**